### PR TITLE
Add protocol driver's customizable configuration

### LINKED ISSUE #78: Create a customizable configuration field and API for users
The SDK currently manages all configuration (local and via the registry). If an author of a device profile wants to add protocol-specific configuration values, this is not possible today. Instead, they would be required handle protocol configuration entirely separate from the existing mechanisms. Fixing this issue would allow a DS develop to leverage the existing configuration mechanisms to support protocol-specific configuration values, perhaps via the use of another level of path in the configuration key naming.

### PR DESCRIPTION
**Pull request by weichou1229**
Fix #78

### CODE REVIEW

File: internal/common/types.go [hunk 1]
```diff
@@ -115,14 +115,16 @@ type Config struct {
...
-	// Device contains device-specific coniguration settings.
+	// Device contains device-specific configuration settings.
 	Device DeviceInfo
 	// Logging contains logging-specific configuration settings.
 	Logging LoggingInfo
 	// Watchers is a map provisionwatchers to be created on startup.
 	Watchers map[string]WatcherInfo
 	// DeviceList is the list of pre-define Devices
 	DeviceList []DeviceConfig `consul:"-"`
+	// Driver is a map for protocol driver's customizable configuration
```
Comment (cloudxxx8) [inline internal/common/types.go]: rephrase
Driver is a string map contains customized configuration for the protocol driver using Device SDK

File: service.go [hunk 2]
```diff
@@ -276,3 +276,7 @@ func NewService(serviceName string, serviceVersion string, confProfile string, c
 func RunningService() *Service {
 	return svc
 }
+
+func DriverProperties() map[string]string {
```
Comment (cloudxxx8) [inline service.go]: rename to `DriverConfigs`

File: internal/common/types.go (hunk 1 above)
Comment (weichou1229) [reply internal/common/types.go]: Rephrased

File: service.go [hunk 3]
```diff
@@ -276,3 +276,7 @@ func NewService(serviceName string, serviceVersion string, confProfile string, c
 func RunningService() *Service {
 	return svc
 }
+
```
Comment (cloudxxx8) [inline service.go]: godoc is needed for public API
// DriverConfigs retrieves the driver specific configuration

**Comment by cloudxxx8** [comment]
@tobiasmo1 @lenny-intel , please see whether this implementation fits your requirement.  thanks

File: service.go (hunk 3 above)
Comment (weichou1229) [reply service.go]: Added

**Comment by tobiasmo1** [comment]
@cloudxxx8 LGTM as well.

**Comment by cloudxxx8** [comment]
Since we encounter the dead DCO issue again, we will open an identical PR tomorrow and merge it directly since it has been reviewed and approve

**Comment by weichou1229** [comment]
recheck

**Comment by weichou1229** [comment]
Reopen PR for dead DCO issue.

**Pull request closed without merging**